tests/applepay_test.py

```diff
@@ -94,35 +149,414 @@ def test_signing_time_equals_current_time(token_fixture):
+        "data": base64.b64encode('sir robin;'),
+        "signature": "does not matter",
+        "header": {
+            "transactionId": binascii.hexlify('sir lancelot;'),
+            "ephemeralPublicKey": base64.b64encode('king arthur;'),
+            "publicKeyHash": "does not matter"
+        }
+    }
+
+    # When the payment data is accessed and decoded
+    payment_data = applepay_utils.get_payment_data(token)
+
+    # Then it is the expected value
+    expected_payment_data = 'king arthur;sir robin;sir lancelot;'
+    assert expected_payment_data == payment_data
+
+
+def test_get_payment_data_includes_application_data():
+    # Given an apple pay token including application data
+    token = {
+        "version": "does not matter",
+        "data": base64.b64encode('sir robin;'),
+        "signature": "does not matter",
+        "header": {
+            "transactionId": binascii.hexlify('sir lancelot;'),
+            "ephemeralPublicKey": base64.b64encode('king arthur;'),
+            "publicKeyHash": "does not matter",
+            "applicationData": binascii.hexlify('sir galahad;')
+        }
+    }
+
+    # When the payment data is accessed and decoded
+    payment_data = applepay_utils.get_payment_data(token)
+
+    # Then it is the expected value
+    expected_payment_data = 'king arthur;sir robin;sir lancelot;sir galahad;'
+    assert expected_payment_data == payment_data
+
+
+def test_get_ber_encoded_signed_attributes(signed_attributes_fixture):
+    # Given: some signed attributes
+    # from the signed_attributes_fixture
+
+    # When the ber-encoded attributes are retrieved
+    signed_attrs_ber = applepay_utils.get_ber_encoded_signed_attributes(signed_attributes_fixture)
+
+    # Then the result is the expected value
+    expected_ber_attributes = '1i0\x18\x06\t*\x86H\x86\xf7\r\x01\t\x031\x0b\x06\t*\x86H\x86\xf7\r\x01\x07\x010\x1c\x06\t*\x86H\x86\xf7\r\x01\t\x051\x0f\x17\r141027195143Z0/\x06\t*\x86H\x86\xf7\r\x01\t\x041"\x04 {M_{\x87\xb5\xfb\n\x11\x9d\xa5w\xa3\xc6\xd9/\xbb\xe6L\xb1\x03\xb2v_M\n\xbe\x0f\xb1\x98\x8er'
+    assert expected_ber_attributes == signed_attrs_ber
+
+
+def test_remove_ec_point_prefix(certificates_fixture):
+    # Given a string that stars with the EC point uncompressed prefix
+    point = '\x04\xc2\x15w\xed'
+
+    # When we remove the ec point prefix from the point string
+    public_key_point = applepay_utils.remove_ec_point_prefix(point)
+
+    # Then the returned bytes are the remaining bytes minus the prefix
+    assert len(public_key_point) == 4
+    assert not public_key_point.startswith("\x04")
+
+
+def test_remove_ec_point_prefix_finds_unexpected_format(certificates_fixture):
+    # Given a string that is not an uncompressed EC point
+    point = 'not.a.point'
+
+    # When we remove the ec point prefix from the point string
+    public_key_bytes = applepay_utils.remove_ec_point_prefix(point)
+
+    # Then no bytes are returned due to the malformed point
+    assert not public_key_bytes
```

Comment:
I will update this to be the value of the point after the uncompressed-indicating byte is removed.